src/Public/OpenApi.ps1

```diff
@@ -146,32 +302,49 @@ function Get-PodeOpenApiDefinition {
         [string]
         $RouteFilter = '/*',
 
+        [Parameter()]
         [switch]
         $RestrictRoutes
     )
 
-    $Title = Protect-PodeValue -Value $Title -Default $PodeContext.Server.OpenAPI.Title
-    if ([string]::IsNullOrWhiteSpace($Title)) {
-        throw 'No Title supplied for OpenAPI definition'
+    $meta = @{
+        RouteFilter    = $RouteFilter
+        RestrictRoutes = $RestrictRoutes
     }
-
-    $Version = Protect-PodeValue -Value $Version -Default $PodeContext.Server.OpenAPI.Version
-    if ([string]::IsNullOrWhiteSpace($Version)) {
-        throw 'No Version supplied for OpenAPI definition'
+    if ($RestrictRoutes) {
+        $meta = @{
+            RouteFilter    = $RouteFilter
+            RestrictRoutes = $RestrictRoutes
+        }
+    } else {
+        $meta = @{}
+    }
+    if ($Title) {
+        $meta.Title = $Title
+    }
+    if ($Version) {
+        $meta.Version = $Version
+    }
+    if ($Description) {
+        $meta.Description = $Description
     }
 
-    $Description = Protect-PodeValue -Value $Description -Default $PodeContext.Server.OpenAPI.Description
+    $oApi = Get-PodeOpenApiDefinitionInternal  -MetaInfo $meta
 
-    # generate the openapi definition
-    return Get-PodeOpenApiDefinitionInternal `
-        -Title $Title `
-        -Version $Version `
-        -Description $Description `
-        -RouteFilter $RouteFilter `
-        -Protocol $WebEvent.Endpoint.Protocol `
-        -Address $WebEvent.Endpoint.Address `
-        -EndpointName $WebEvent.Endpoint.Name `
-        -RestrictRoutes:$RestrictRoutes
+    switch ($Format) {
+        'Json' {
+            return ConvertTo-Json -InputObject $oApi -Depth 10
+        }
+        'Json-Compress' {
+            return ConvertTo-Json -InputObject $oApi -Depth 10 -Compress
+        }
+        'Yaml' {
+            return ConvertTo-PodeYaml -InputObject $oApi -Depth 10
+        }
+        Default {
+            return $oApi
+        }
+    }
```

Comment:
The `$Format` in the switch will need a `.ToLower()`, and all the cases converting to lowercase (ie: `Json` to `json`, etc.). If somebody does pass `json` instead of `Json` this is what the Format parameter will be.